File: Archive/Matchismo/Model/Card.m
//
//  Card.m
//  Matchismo
//
//  Created by Hector Enrique Gomez Morales on 6/15/13.
//  Copyright (c) 2013 Hector Enrique Gomez Morales. All rights reserved.
//

#import "Card.h"

@implementation Card

- (int)match:(NSArray *)otherCards
{
    int score = 0;
    
    for (Card *card in otherCards) {
        if ([card.contents isEqualToString:self.contents]) {
            score = 1;
        }
    }
    
    return score;
}

@end

File: Matchismo/Model/Card.swift
//
//  Card.swift
//  Matchismo
//
//  Created by Hector Enrique Gomez Morales on 6/17/14.
//  Copyright (c) 2014 Hector Enrique Gomez Morales. All rights reserved.
//

import Foundation

class Card {
    var chosen = false
    var matched = false
    
    func match(otherCards: Card[]) -> Int {
        var score = 0
        
        for card in otherCards {
            if card.contents() == self.contents() {
                score = 1
            }
        }
        
        return score
    }
    
    func contents() -> String {
        return "123"
    }
}